lib/module-types.js

```diff
@@ -0,0 +1,51 @@
+module.exports = (configuredExtensions, defaultModuleType) => {
+	const lockedModuleTypes = {
+		cjs: 'commonjs',
+		mjs: 'module',
+		js: defaultModuleType
+	};
+
+	const createModuleTypesFromExtensionsObject = extensionsObject => {
+		const configuredModuleTypes = {};
+
+		for (const [extension, type] of Object.entries(extensionsObject)) {
+			if (extension in lockedModuleTypes && type !== lockedModuleTypes[extension]) {
+				throw new TypeError(`.${extension} files can only be configured as ’${lockedModuleTypes[extension]}’${
+					extension === 'js' ?
+						` when the nearest parent package.json ${type === 'module' ? 'does not contain' : 'contains'} "type": "module"` : ''
+				}, found ’${type}’ instead.`);
```

Comment:
Rather than requiring users to specify the correct type for the standard extensions, perhaps we can let users set them to `true` (and nothing else). That way you do not need to change your AVA configuration if you change the package type, nor do you need to be needlessly explicit. What do you think?

Reply:
Yeah, I think it might work, I don't see any drawbacks with that. Should it throw if it's something else (`false | string`)? Should other extensions still only allow `'commonjs' | 'module'`?

Reply:
> Should it throw if it's something else (`false | string`)?

Yes, let's do that.

> Should other extensions still only allow `'commonjs' | 'module'`?

Indeed.